enterprise/backend/src/metabase_enterprise/transforms/models/transform_run.clj

```diff
@@ -171,52 +192,42 @@
-                           (conj [:and
-                                  [:in :transform_id transform_ids]
-                                  [:in :transform_id {:select [:transform_id]
-                                                      :from   [:transform_transform_tag]
-                                                      :where  [:in :tag_id transform_tag_ids]}]])
-
-                           ;; Only transform_ids
-                           (and (seq transform_ids) (not (seq transform_tag_ids)))
+        where-cond       (cond-> []
+                           (some? start_time)
+                           (conj (timestamp-constraint :start_time start_time))
+
+                           (some? end_time)
+                           (conj (timestamp-constraint :end_time end_time))
+
+                           (some? run_method)
+                           (conj [:= :run_method run_method])
+
+                           (seq transform_ids)
                            (conj [:in :transform_id transform_ids])
 
-                           ;; Only transform_tag_ids
-                           (and (seq transform_tag_ids) (not (seq transform_ids)))
+                           (seq transform_tag_ids)
                            (conj [:in :transform_id {:select [:transform_id]
                                                      :from   [:transform_transform_tag]
                                                      :where  [:in :tag_id transform_tag_ids]}])
 
-                           ;; statuses condition
                            (seq statuses)
                            (conj [:in :status statuses])
 
-                           ;; is_active condition for started status
-                           (and (seq statuses)
-                                (some #(= % "started") statuses))
+                           ;; optimization: is_active condition for started status
+                           (and (= (first statuses) "started")
+                                (nil? (next statuses)))
                            (conj [:= :is_active true]))
-
-        where-clause  (when (seq where-conditions)
-                        (if (= 1 (count where-conditions))
-                          (first where-conditions)
-                          (into [:and] where-conditions)))
-        query-options (cond-> {:order-by order-by
-                               :offset offset
-                               :limit    limit}
-                        where-clause (assoc :where where-clause))
-        runs          (t2/select :model/TransformRun query-options)
-        count-options (cond-> {}
-                        where-clause (assoc :where where-clause))]
-
+        where-clause     (when (seq where-cond)
+                           (into [:and] where-cond))
+        count-options    (m/assoc-some {} :where where-clause)
+        query-options    (merge {:order-by order-by :offset offset :limit limit}
+                                count-options)
```

Comment:
it's the logic of the paging that `query-options` is an extension of the `count-options`.